flink-ml-lib/src/main/java/org/apache/flink/ml/common/gbt/GBTRunner.java

```diff
@@ -0,0 +1,247 @@
+                        });
+
+        DataStream<Row> data = tEnv.toDataStream(dataTable);
+        final IterationID iterationID = new IterationID();
+        DataStreamList dataStreamList =
+                Iterations.iterateBoundedStreamsUntilTermination(
+                        DataStreamList.of(initTrainContext.broadcast()),
+                        ReplayableDataStreamList.notReplay(data, featureMeta),
+                        IterationConfig.newBuilder()
+                                .setOperatorLifeCycle(IterationConfig.OperatorLifeCycle.ALL_ROUND)
+                                .build(),
+                        new BoostIterationBody(iterationID, p));
+        return dataStreamList.get(0);
+    }
+
+    public static GbtParams fromEstimator(BaseGBTParams<?> estimator, TaskType taskType) {
+        final Map<Param<?>, Object> paramMap = estimator.getParamMap();
+        final Set<Param<?>> unsupported =
+                new HashSet<>(
+                        Arrays.asList(
+                                BaseGBTParams.WEIGHT_COL,
+                                BaseGBTParams.LEAF_COL,
+                                BaseGBTParams.VALIDATION_INDICATOR_COL));
+        List<Param<?>> unsupportedButSet =
+                unsupported.stream()
+                        .filter(d -> null != paramMap.get(d))
+                        .collect(Collectors.toList());
+        if (!unsupportedButSet.isEmpty()) {
+            throw new UnsupportedOperationException(
+                    String.format(
+                            "Parameters %s are not supported yet right now.",
+                            unsupportedButSet.stream()
+                                    .map(d -> d.name)
+                                    .collect(Collectors.joining(", "))));
+        }
+
+        GbtParams p = new GbtParams();
+        p.taskType = taskType;
+        p.featureCols = estimator.getInputCols();
+        p.vectorCol = estimator.getFeaturesCol();
+        p.isInputVector = (null == p.featureCols);
+        p.labelCol = estimator.getLabelCol();
+        p.weightCol = estimator.getWeightCol();
+        p.categoricalCols = estimator.getCategoricalCols();
+
+        p.maxDepth = estimator.getMaxDepth();
+        p.maxBins = estimator.getMaxBins();
+        p.minInstancesPerNode = estimator.getMinInstancesPerNode();
+        p.minWeightFractionPerNode = estimator.getMinWeightFractionPerNode();
+        p.minInfoGain = estimator.getMinInfoGain();
+        p.maxIter = estimator.getMaxIter();
+        p.stepSize = estimator.getStepSize();
+        p.seed = estimator.getSeed();
+        p.subsamplingRate = estimator.getSubsamplingRate();
+        p.featureSubsetStrategy = estimator.getFeatureSubsetStrategy();
+        p.validationTol = estimator.getValidationTol();
+        p.gamma = estimator.getRegGamma();
+        p.lambda = estimator.getRegLambda();
+
+        if (TaskType.CLASSIFICATION.equals(p.taskType)) {
```

Comment:
Would it be simpler to use `estimator instanceof GBTClassifier`  instead of reading/writing an extra parameter `taskType` here?

Same for other parameters and methods (e.g. `trainClassifier()`) which can be replaced with the `instanceof`.